homeassistant/components/tplink/number.py

```diff
@@ -0,0 +1,97 @@
+"""Support for TPLink number entities."""
+
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from kasa import Feature
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .entity import (
+    CoordinatedTPLinkFeatureEntity,
+    TPLinkFeatureEntityDescription,
+    async_refresh_after,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TPLinkNumberEntityDescription(
+    NumberEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+NUMBER_DESCRIPTIONS: Final = (
+    TPLinkNumberEntityDescription(
+        key="smooth_transition_on",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="smooth_transition_off",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="auto_off_minutes",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="temperature_offset",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="target_temperature",
+        mode=NumberMode.BOX,
+    ),
```

Comment:
I cannot seem to find the mapping / where the device class for this one is set?

Could you point me to it?

Reply:
Good catch! This was added mistakenly, it's something the thermostat uses & can be removed as the climate platform will use it on its own.

Reply:
Removed.